### ISSUE TITLE
DOCS: RAII style `NoGIL` class

### ISSUE THREAD

**Issue by jakirkham** [wjakob/pybind11#22]
This would be pretty easy to add, but would be quite nice to have by default (if not already present). Adding an RAII style object that releases the GIL when constructed and acquires the GIL when destructed. Having this object would make it easy to have blocks of C++ code that can run in parallel to the Python interpreter.

**Comment by jakirkham** [comment]
Somehow I completely missed this ( https://github.com/wjakob/pybind11/blob/8a34f70f83a782029c392f9d576043bcd4b16445/include/pybind11/pybind11.h#L902-L916 ). Maybe something brief should be added to the docs about this.

**Comment by wjakob** [comment]
yep, no problem. I just committed some docs for this function.

**Comment by jakirkham** [comment]
Thanks.
